File: protobuf-serialization/src/main/java/com/stannis/benchmark/repository/RecipeRepository.java
package com.stannis.benchmark.repository;

import com.stannis.benchmark.model.Recipe;
import generated.stannis.benchmark.Messages;


public interface RecipeRepository {

    Messages.Recipe getProtoRecipeById(int id);

    Integer save(Messages.Recipe recipe);

    Recipe getRecipeById(int id);

    Integer save(Recipe recipe);
}

File: protobuf-bench/src/main/kotlin/com/stannis/protobufbench/repository/RecipeRepository.kt
package com.stannis.protobufbench.repository

import com.stannis.protobufbench.model.Recipe
import generated.stannis.proto.Messages


interface RecipeRepository {

    fun getProtoRecipeById(id: Int): Messages.Recipe?

    fun save(recipe: Messages.Recipe?): Int?

    fun getRecipeById(id: Int): Recipe?

    fun save(recipe: Recipe?): Int?

}
